Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -49,7 +49,7 @@ extension Style {
         for sourceInfo in vectorSources {
             if locale.identifier.starts(with: "zh") {
                 // Force unwrapping since `allSourceIdentifiers` is getting a fresh list of valid sources
-                let vectorSource = try! source(withId: sourceInfo.id) as VectorSource
+                let vectorSource = try! source(withId: sourceInfo.id, type: VectorSource.self) as VectorSource
```

Comment:
Here's another one where we need to remove "as". Could you do a careful self-review of this PR to find the rest of these cases and cases where the variable being assigned is unnecessarily given an explicit type?